Old OpenLane version can pass the detail routing but latest version cannot

### Description
Hello, with the same design, I happened to find that the old openlane version can pass the detail routing which will make the violations to 0, while the latest openlane cannot pass and remain several violations. I checked the openroad-app commit as well. The old openlane's openroad-app commit is `0b8b7ae255f8fbbbefa57d443949b84e73eed757` while the latest openlane's openroad commit is `e036ecfaca4bc0efe88a54085efcf0f562c48a6b`. Therefore, I suspect that the issue should be caused by the openroad-app, and believe the mpw-7a commit can work while the other commit cannot. If it is true, how can I find the mpw-7a based openlane and/or switch to mpw-7a with latest openlane? Simply changing the commit in openroad-app with the mpw-7a commit in the latest openlane cannot work and it will warn me to change the environment. I do not have any idea about how to change the environment. Could anyone can point me in the right direction? Thanks!

### Environment
The old environment
```
python3 ./env.py issue-survey
Kernel: Linux v4.15.0-191-generic
Distribution: ubuntu 18.04
Python: v3.6.9 (OK)
Container Engine: docker v20.10.7 (OK)
OpenLane Git Version: a9883335a102c9ba1d862a209a72fc8b4c20c785
pip: INSTALLED
pip:venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

a988333 2022-06-18T11:53:39+02:00 Enhancements + Bugfixes to Multi-PDK Support (#1138) - Mohamed Gaber -  (HEAD -> master, tag: 2022.06.19_01.37.45)
732c856 2022-06-18T11:22:57+02:00 Fix `skip_basic_prep`, step mechanics (#1143) - Mohamed Gaber -  ()
074a92b 2022-06-14T15:01:11+02:00 Fix Magic Antenna Check (#1142) - Mohamed Gaber -  (tag: 2022.06.15_01.35.48)
```

The latest environment
```
python3 ./env.py issue-survey
Kernel: Linux v4.15.0-191-generic
Distribution: ubuntu 18.04
Python: v3.7.13 (OK)
Container Engine: docker v20.10.17 (OK)
OpenLane Git Version: 4476a58407d670d251aa0be6a55e5391bb181c4e
pip: INSTALLED
python-venv: INSTALLED
---
PDK Version Verification Status: OK
---
Git Log (Last 3 Commits)

4476a58 2022-07-29T16:03:10+02:00 Sky130 Enhancements (#1240) - Mohamed Gaber -  (HEAD -> master, tag: 2022.07.30, origin/master, origin/HEAD)
4f63c8c 2022-07-27T17:26:27+02:00 Various CI Fixes (#1235) - Mohamed Gaber -  (tag: 2022.07.28)
0b55101 2022-07-26T20:37:20+02:00 [BOT] Update openroad_app (#1219) - Openlane Bot -  (tag: 2022.07.27)
---
Git Remotes

origin  https://github.com/The-OpenROAD-Project/OpenLane.git (fetch)
origin  https://github.com/The-OpenROAD-Project/OpenLane.git (push)
```

### Reproduction Material
No `issue_repoduction` directory was produced. Will upload all other files if needed.

### Expected behavior
The design should pass detailed routing without violation

### Logs
With latest version
```
[ERROR]: There are violations in the design after detailed routing.
[ERROR]: Total Number of violations is 18
[ERROR]: Flow failed.
```

Thanks for the help. I tried to re-build the OpenLane. However, I meet this error when I use `make` in the OpenLane directory after `git-clone` the tool.

```
make[1]: Entering directory '/home/htf6ry/OpenLane'
57f3f21d6e1d0403a88d7a7973f7447597aae160: Pulling from efabless/openlane
Digest: sha256:0b9a27fe51d8d709e77f653c39dd1244738885d0d39fdefd81a272824a33b46f
Status: Image is up to date for efabless/openlane:57f3f21d6e1d0403a88d7a7973f7447597aae160
docker.io/efabless/openlane:57f3f21d6e1d0403a88d7a7973f7447597aae160
make[1]: Leaving directory '/home/htf6ry/OpenLane'
rm -rf ./venv
python3 -m venv ./venv
Error: Command '['/home/htf6ry/OpenLane/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
Makefile:130: recipe for target 'venv/created' failed
make: *** [venv/created] Error 1
```

I do not have any idea what happened....

--- a/.github/workflows/tool_updater.yml
+++ b/.github/workflows/tool_updater.yml
@@ -29,7 +29,7 @@ jobs:
 
       - name: Create Pull Request
         if: ${{ env.NO_UPDATE != '1' }}
-        uses: peter-evans/create-pull-request@v3
+        uses: peter-evans/create-pull-request@v4
         with:
           token: ${{ secrets.MY_TOKEN }}
           title: "[BOT] Update ${{ env.TOOL }}"
@@ -61,7 +61,7 @@ jobs:
 
       - name: Create Pull Request
         if: ${{ env.NO_UPDATE != '1' }}
-        uses: peter-evans/create-pull-request@v3
+        uses: peter-evans/create-pull-request@v4
         author: ${{ secrets.BOT_AUTHOR_LINE }}
         committer: ${{ secrets.BOT_AUTHOR_LINE }}
         with:

--- a/dependencies/tool_metadata.yml
+++ b/dependencies/tool_metadata.yml
@@ -9,7 +9,7 @@
     make install
 - name: magic
   repo: https://github.com/rtimothyedwards/magic
-  commit: f7df5e7c86fb47c5fd445c846afddc6fbabad6ae
+  commit: 5d51e10fb969b31e6e95b5fb78d21efeccc73c14
   build: |
     ./configure --prefix=$PREFIX $MAGIC_CONFIG_OPTS
     make clean
@@ -58,7 +58,7 @@
   in_install: false
 - name: openroad_app
   repo: https://github.com/The-OpenROAD-Project/OpenROAD
-  commit: 4a99e88667b0840531ca0096c4fa0da8f80d6cb1
+  commit: 4174c3ad802d2ac1d04d387d2c4b883903f6647e
   build: ""
   in_install: false
 - name: git

--- a/scripts/magic/drc.tcl
+++ b/scripts/magic/drc.tcl
@@ -22,7 +22,7 @@ if { [info exist ::env(MAGIC_DRC_USE_GDS)] && $::env(MAGIC_DRC_USE_GDS) } {
 			lef read $lef_file
 		}
 	}
-	def read $::env(CURRENT_DEF)
+	def read $::env(CURRENT_DEF) -noblockage
 }
 
 set drc_rpt_path $::env(drc_prefix).rpt

